### ISSUE TITLE
Kodi 18 Leia alpha-version issues

### ISSUE THREAD

**Issue by troych** [croneter/PlexKodiConnect#339]
Hi there, I'm getting this error when I enable the music library in PlexKodiConnect's options on my Kodi 18 machine. Any idea what the cause of this may be? I'm running native paths of this helps anything.

Kodi 18.0-ALPHA1 Git:20170903-7161dc0
PlexKodiConnect 1.8.11
MyMusic67.db

```
14:03:48.449 T:140426425100032  NOTICE: PLEX.utils -> ERROR: add_updateSong has crashed. Error: no such table: song
14:03:48.497 T:140426425100032  NOTICE: PLEX.utils -> ERROR: Traceback:
                                            Traceback (most recent call last):
                                              File "/home/kodi/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/utils.py", line 969, in wrapper
                                                return func(*args, **kwargs)
                                              File "/home/kodi/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/itemtypes.py", line 1590, in add_updateSong
                                                kodicursor.execute("select coalesce(max(idSong),0) from song")
```

**Comment by croneter** [comment]
As Kodi 18 is still in alpha, this could be "fixed" or changed again on the Kodi side any day. 

Don't expect a quick fix from the PKC side...

**Comment by troych** [comment]
I see, thanks for the info. I'll close t this and disable updates for PKC for now, lets hope 18 is stabilizing soon. 

For anybody with the same problem: I've manually replaced the music db version number (67) in variables.py and strArtists with strArtistDisp in itemtypes.py. Things seem to be working for now.

I just can't go back to a world without netflix plugin for kodi. :D

**Comment by croneter** [comment]
If you run into SSL issues like
```
ERROR: CCurlFile::Stat - Failed: SSL connect error(35) for https://192.168.2.200:32400/photo/:/transcode?width=4000&height=4000&minSize=1&upscale=0&url=/library/metadata/7977/art/1477376189&X-Plex-Token=<TOKEN>
ERROR: CCurlFile::FillBuffer - Failed: SSL connect error(35)
ERROR: CCurlFile::Open failed with code 0 for https://192.168.2.200:32400/photo/:/transcode/?width=4000&height=4000&minSize=1&upscale=0&url=/library/metadata/7977/art/1477376189&X-Plex-Token=<TOKEN>
ERROR: GetDirectory - Unable to get http directory (https://192.168.2.200:32400/photo/:/transcode/?width=4000&height=4000&minSize=1&upscale=0&url=/library/metadata/7977/art/1477376189&X-Plex-Token=<TOKEN>)
ERROR: GetDirectory - Error getting https://192.168.2.200:32400/photo/:/transcode/?width=4000&height=4000&minSize=1&upscale=0&url=/library/metadata/7977/art/1477376189&X-Plex-Token=<TOKEN>
```
then try this:
login to https://app.plex.tv/desktop and connect to your PMS. Then navigate to any movie, hit the three dots ... in the upper right corner, then Show Information, then Show XML. In the new browser window, copy the url like
```
https://192-168-0-2.6eb604e9b8c94479bfc096<CUT>.plex.direct
```
In the PKC connection settings, manually connect to that URL

**Comment by croneter** [comment]
The PKC library sync thread might crash with the following log entry:
```
OperationalError: no such table: art
```
Unfortunately, a lot of code references this table. I'm not going to adapt the PKC code until it's clear that this change is permanent - so until Kodi 18 is at least beta

**Comment by Lou-Cipher** [comment]
I just wanted to leave a quick comment as to what I had to do to get all my media working and no crashes with Kodi 18 and the current 1.8.18 version of plexkodiconnect.  Some db modifications were necessary, and I'm going to leave step-by-step directions for anyone who might not be familiar with sqlite editing.  This works on Kodi 18 built from master as of about a half hour ago.  Note that this is a bit hackity, and adds DB tables that plexkodiconnect expects to exist, but which do not, as of this writing, exist in the DBs.  This may well break things in the future, and hopefully be unnecessary soon enough (and many thanks to croneter for all his hard work on this amazing plugin! And troych for putting me on the path to this solution), but for now this will get you going.

Go to ~/.kodi/userdata/Databases (or wherever they are on your OS, linux ftw)
Run sqlite3 (install if necessary), and run the following commands:

.open MyVideos109.db
CREATE TABLE tag (tag_id integer primary key, name integer);
.open MyMusic69.db
CREATE TABLE album_genre (iOrder INTEGER, idGenre INTEGER, idAlbum INTEGER, FOREIGN KEY(idGenre) REFERENCES genre(idGenre), FOREIGN KEY(idAlbum) REFERENCES album(idAlbum));
CREATE TABLE albuminfosong (idAlbumInfoSong INTEGER PRIMARY KEY, idAlbumInfo INTEGER, iTrack INTEGER, strTitle TEXT, iDuration INTEGER, FOREIGN KEY(idAlbumInfo) REFERENCES album(idAlbumInfo));
.exit

Now edit ~/.kodi/addons/plugin.video.plexkodiconnect/resources/lib/variables.py

Locate _DB_MUSIC_VERSION, and ensure that Kodi version 18 (with the comment # Leia) is set to DB version 69.  Then, locate _DB_VIDEO_VERSION, and ensure that Kodi version 18 is set to DB version 109.

Restart Kodi, and resync your database via plexkodiconnect's Advanced options menu.  Should now have full sync for Movies, TV Shows, and Music (although Music still requires direct paths to avoid the dreaded 401 endless sync, see issue #14). Should be good to go.

**Comment by fkmclane** [comment]
I believe I have this working with the database changes for Leia. I basically removed code that updated the database for things Kodi didn't even use anyway and bumped the DB versions. I'll link the relevant Kodi commits later showing why they were removed. See the [fkmclane/PlexKodiConnect](https://github.com/fkmclane/PlexKodiConnect/tree/hotfixes) hotfixes branch.

**Comment by hanzoh** [comment]
The jump from 108 to 109 only removed a column in the settings table: https://github.com/xbmc/xbmc/commit/9dc6e1dbe0ecd5d0c71ec983bcc95c765df098b0#diff-dbf978b6cac348a2af93d3beec420e43

So I am using 1.8.18 just fine with the latest nightly by changing 108 to 109 in variables.py.

Regarding the warning about **cannot reference separator setting in old setting condition "eq(-1,true)" for "skipContextMenu"**:
I think the Kodi warning is just right, because this in settings.xml cannot work:

`<setting type="lsep" label="" />
<setting id="skipContextMenu" type="bool" label="30520" default="false" visible="eq(-1,true)" subsetting="true" />
`

visible="eq(-1,true)" determines the visibility of a setting by the state of the prior settings element. That element is a separator which has no state. So the "visible" tag can be removed as the default is 'true'.

**Comment by hanzoh** [comment]
@croneter I have just tested the beta release 2.0.4 with Kodi 18.
With the 2 fixes I proposed above it also works fine except for one thing:
When starting any movie episode, I get a popup that playback has failed and immediately after that it starts the video, but the box remains and I have to hit ok.

I checked playback.py and found the attribute RESOLVE which also triggers playing back PKC_Dummy_Path_Which_Fails before the actual video. Are you planning on making this configurable because right now it is hardcoded with true in all instances?
